_inc/lib/core-api/wpcom-fields/attachment-fields-videopress.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Extend the REST API functionality for VideoPress users.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Add per-attachment VideoPress data.
+ *
+ * { # Attachment Object
+ *   ...
+ *   jetpack_videopress_guid: (string) VideoPress identifier
+ *   ...
+ * }
+ *
+ * @since 7.1.0
+ */
+class WPCOM_REST_API_V2_Attachment_VideoPress_Field extends WPCOM_REST_API_V2_Field_Controller {
+	/**
+	 * The REST Object Type to which the jetpack_videopress_guid field will be added.
+	 *
+	 * @var string
+	 */
+	protected $object_type = 'attachment';
+
+	/**
+	 * The name of the REST API field to add.
+	 *
+	 * @var string $field_name
+	 */
+	protected $field_name = 'jetpack_videopress_guid';
+
+	/**
+	 * Registers the jetpack_videopress field and adds a filter to remove it for attachments that are not videos.
+	 */
+	public function register_fields() {
+		parent::register_fields();
+
+		add_filter( 'rest_prepare_attachment', array( $this, 'remove_field_for_non_videos' ), 10, 2 );
+	}
+
+	/**
+	 * Defines data structure and what elements are visible in which contexts
+	 */
+	public function get_schema() {
+		return array(
+			'$schema'     => 'http://json-schema.org/draft-04/schema#',
+			'title'       => $this->field_name,
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'readonly'    => true,
+			'description' => __( 'Unique VideoPress ID', 'jetpack' ),
+		);
+	}
+
+	/**
+	 * Getter: Retrieve current VideoPress data for a given attachment.
+	 *
+	 * @param mixed           $object  Response from the attachment endpoint.
+	 * @param WP_REST_Request $request Request to the attachment endpoint.
+	 *
+	 * @return string
+	 */
+	public function get( $object, $request ) {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$blog_id = get_current_blog_id();
+		} else {
+			$blog_id = Jetpack_Options::get_option( 'id' );
+		}
+
+		$post_id = absint( $object['id'] );
+
+		$videopress_guid = $this->get_videopress_guid( $post_id, $blog_id );
+
+		if ( ! $videopress_guid ) {
+			return '';
+		}
+
+		return $videopress_guid;
+	}
+
+	/**
+	 * Gets the VideoPress GUID for a given attachment.
+	 *
+	 * This is pulled out into a separate method to support unit test mocking.
+	 *
+	 * @param int $attachment_id Attachment ID.
+	 * @param int $blog_id Blog ID.
+	 *
+	 * @return string
+	 */
+	public function get_videopress_guid( $attachment_id, $blog_id ) {
+		return video_get_info_by_blogpostid( $blog_id, $attachment_id )->guid;
+	}
+
+	/**
+	 * Checks if the given attachment is a video.
+	 *
+	 * @param object $attachment The attachment object.
+	 *
+	 * @return false|int
+	 */
+	public function is_video( $attachment ) {
+		return preg_match( '@^video/@', $attachment->post_mime_type );
```

Comment:
Curious why you switched to `preg_match` from the initial suggestion of `strpos`? The latter is considered [[ http://php.net/manual/en/function.preg-match.php#refsect1-function.preg-match-notes | more performant and preferable ]] when checking for a string within a string. I'm also not familiar with the `@`, nor could I quickly find a reference; what is its purpose?

Reply:
The `@` is just the delimiter. In PCRE (what PHP uses for regex's, hence: **p**reg...), you can use any most any "special" character: http://php.net/manual/en/regexp.reference.delimiters.php

We tend to use `/` for consistency, but many characters will do. Here, it's convenient to use something that is not `/` since, otherwise, we'd have to escape the `/` in the pattern itself.

If the switch to `preg_match()` was made for readability, we can also use `wp_startswith( $haystack, $needle )` in WP.com and Jetpack, which may meet the readability requirement while still using the more performant `strpos()`.

Reply:
concerning the `@` PHP is pretty permissive in which [delimiters](http://php.net/manual/en/regexp.reference.delimiters.php) it uses for RegExp patterns - well, this probably goes back to the PERL roots

you can basically use almost anything there as long as it balances on the end.

also, the transformed version of this match…

```php
return 0 === strpos( $attachment->post_mime_type, 'video/' );
```

Reply:
> Curious why you switched to `preg_match` from the initial suggestion of `strpos`?

Just for readability and because I noted we are already using that in other places ([#](https://github.com/Automattic/jetpack/blob/7b7d869dbaa6131e8782431c0f78638cc990ea98/modules/videopress/class.jetpack-videopress.php#L302), [#](https://github.com/Automattic/jetpack/blob/7b7d869dbaa6131e8782431c0f78638cc990ea98/json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php#L167)).

Didn't know about the existence of `wp_startswith` (thanks @mdawaffe!) so I'll commit now a change using it :).

Reply:
> In the current implementation, it's added for all videos. Is that the desired behavior?

That's right. 

From an API consumer perspective, I think it is more natural to have the field in all videos, since all of them can potentially be VideoPress videos. Therefore, the value of this field will indicate if it is a VideoPress video or not.

This also makes the implementation easier because we don't need to check if the field is present or not.